### DISCUSSION TITLE
Docker-Installation

### DISCUSSION THREAD

**Discussion by synchronierer** [jzakotnik/openlibry#80] [Q&A]
Als Docker-Dummy habe ich einige Fragen an den Entwickler gestellt und hier teilen wir unsere Erfahrungen. :-)

**Comment by jzakotnik** [comment]
Für Issues mit dem Docker Container, falls dieser direkt über http und eine interne IP angesprochen wurden. In dem Fall sind die Content Security Policies zu eng - für den Test kann man sie über das `.env` File mit diesem Flag ausschalten:
```
# Configure if content security policy is set on the http headers, if you don't want CSR header, use "insecure"
SECURITY_HEADERS=insecure
```

**Reply by synchronierer** [reply]
Works like a charm...
Links sind jetzt anklickbar, mit /auth/register kann ich einen Benutzer registrieren. Allerdings wird der mir dann nicht unter "Nutzer" angezeigt.

Hier aber mal alle Schritte wie ich bisher vorgegangen bin:
1.	Im Homeverzeichnis ausführen:
sudo git clone https://github.com/jzakotnik/openlibry
2.	In das Arbeitsverzeichnis wechseln:
cd openlibry
3.	Das Enviroment-File anpassen (wird nur mit ls -a aufgeführt):
sudo cp .env_example .env
sudo nano .env
Der Wert AUTH_ENABLED sollte zunächst auf false gesetzt werden, so dass man sich beim ersten Login ohne Authentifizierung einloggen kann. Via https://<domain>/auth/register kann man dann den ersten User anlegen und danach AUTH_ENABLED auf true setzen. (Die Ports habe ich alle auf 3000 gelassen.) WANN WIRKEN SICH DIESE ÄNDERUNGEN AUS?
4.	Build-Prozess starten (dauert einige Zeit):
docker buildx build --no-cache -t openlibry .
5.	Das Image ausführen:
docker compose up -d
6.	Im Container eine Bash aufrufen und das DB-File erzeugen, das als Volume gemappt wird:
docker exec -it openlibry /bin/sh
npx prisma db push
exit
7.	OpenLibry im Browser öffnen:
https://<<IP_des_Server>>:3000
8.    Einen Admin-User über /auth/register erstellen.
9.   WIE GEHT ES WEITER? :-)

**Comment by jzakotnik** [comment]
Vorschlag, ein Video wie man quasi from scratch anfängt, könnte das als Screencast aufnehmen..  Hier ein Issue dafür #81 

Der Admin User selbst ist typischerweise kein Nutzer des Systems, z.B. wenn es Ehrenamtliche sind, welche die Bücherei betreiben, dann leihen sie selbst keine Bücher aus. Die Nutzer sind Lehrkräfte oder die SuS.

Also Nächstes würde man neue Nutzer anlegen oder (was bisher immer passiert ist) aus [OpenBiblio importieren](https://github.com/jzakotnik/openlibry/blob/main/doc/OpenBiblioImport.md).

**Reply by synchronierer** [reply]
Klar, das kann ich gerne angehen... ich versuche gerade den Import der alten OBiblio-Daten. Wenn ich das auch noch geschafft habe, könnte ich ja ans Video gehen.
